src/core/SurfaceNode.cc

```diff
@@ -262,17 +263,18 @@ const Geometry *SurfaceNode::createGeometry() const
     double v3 = data[ (columns - 1) + (i - 1) * columns ];
     double v4 = data[ (columns - 1) + (i) * columns ];
 
-    p->append_poly(4);
-    p->append_vertex(ox + 0, oy + i - 1, min_val);
-    p->append_vertex(ox + 0, oy + i - 1, v1);
-    p->append_vertex(ox + 0, oy + i, v2);
-    p->append_vertex(ox + 0, oy + i, min_val);
-
-    p->append_poly(4);
-    p->insert_vertex(ox + columns - 1, oy + i - 1, min_val);
-    p->insert_vertex(ox + columns - 1, oy + i - 1, v3);
-    p->insert_vertex(ox + columns - 1, oy + i, v4);
-    p->insert_vertex(ox + columns - 1, oy + i, min_val);
+
+    ind1=builder.vertexIndex(Vector3d(ox + 0, oy + i - 1, min_val));
+    ind2=builder.vertexIndex(Vector3d(ox + 0, oy + i - 1, v1));
+    ind3=builder.vertexIndex(Vector3d(ox + 0, oy + i, v2));
+    ind4=builder.vertexIndex(Vector3d(ox + 0, oy + i, min_val));
+    builder.append_poly({ind1,ind2,ind3,ind4});
+
+    ind1=builder.vertexIndex(Vector3d(ox + columns - 1, oy + i, min_val));
+    ind2=builder.vertexIndex(Vector3d(ox + columns - 1, oy + i, v4));
+    ind3=builder.vertexIndex(Vector3d(ox + columns - 1, oy + i - 1, v3));
+    ind4=builder.vertexIndex(Vector3d(ox + columns - 1, oy + i - 1, min_val));
+    builder.append_poly({ind1,ind2,ind3,ind4});
```

Comment:
Some whitespace is needed in various places. Could you try running our clang-format on this?
I cannot remember exactly how it works but ask in IRC :)